sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -11222,6 +11216,47 @@ public void testOrderByAlongWithAliasOrderByTimeGroupByMulti()
     );
   }
 
+  @Test
+  public void testOrderByAlongWithInternalScanQuery()
+  {
+    testQuery(
+        "select __time as t, m1 from druid.foo where (m1 in (select distinct m1 from druid.foo)) order by 1 limit 1",
```

Comment:
Thanks for the new test. The PR description provides a different query. However, I see no test for that specific query. Does such a test already exist? If so, then this change should have required a change to the expected results. Else, should we add a test for the actual case that this PR claims to fix?

Reply:
I'll add a test mimicking the actual query